examples/step-69/step-69.cc

```diff
@@ -0,0 +1,2926 @@
+#include <deal.II/base/quadrature.h>
+#include <deal.II/base/timer.h>
+#include <deal.II/base/work_stream.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_renumbering.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/mapping.h>
+#include <deal.II/fe/mapping_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/manifold_lib.h>
+
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+#include <deal.II/lac/la_parallel_vector.h>
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/sparse_matrix.templates.h>
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/meshworker/scratch_data.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/vector_tools.h>
+
+// In addition to above deal.II specific includes, we also include four
+// boost headers. The first two are for binary archives that we will use
+// for implementing a check-pointing and restart mechanism.
+#include <boost/archive/binary_iarchive.hpp>
+#include <boost/archive/binary_oarchive.hpp>
+
+// The last two boost header files are for creating custom iterator ranges
+// over integer intervals.
+#include <boost/range/irange.hpp>
+#include <boost/range/iterator_range.hpp>
+
+// @sect3{Class template declarations}
+//
+// We begin our actual implementation by declaring all classes with its
+// data structures and methods upfront. In contrast to previous example
+// steps we use a more fine-grained encapsulation of concepts, data
+// structures, and parameters into individual classes. A single class thus
+// usually centers around either a  single data structure (such as the
+// Triangulation) in the <code>Discretization</code> class, or a single
+// method (such as the <code>step()</code> function of the
+// <code>TimeStep</code> class). We typically declare parameter variables
+// and scratch data object private and make methods and data structures
+// used by other classes public.
+//
+// @note A cleaner approach would be to guard access to all data
+// structures by <a
+// href="https://en.wikipedia.org/wiki/Mutator_method">getter/setter
+// functions</a>. For the sake of brevity, we refrain from that approach,
+// though.
+//
+// We also note that the vast majority of classes is derived from
+// ParameterAcceptor. This facilitates the population of all the global
+// parameters into a single (global) ParameterHandler. More explanations
+// about the use inheritance from ParameterAcceptor as a global subscription
+// mechanism can be found in Step-59.
+
+namespace Step69
+{
+  using namespace dealii;
+
+  // We start with an enum describing all possible boundary conditions
+  // encountered in this tutorial step. Such an enum allows us to refer to
+  // boundary types by a mnemonic (such as
+  // <code>Boundary::do_nothing</code>) rather than a numerical value.
+  enum Boundary : types::boundary_id
```

Comment:
who not use an `enum class`?

Reply:
Because an `enum class` declares its own type and that would necessitate to `reinterpret_cast` the `Boundary` type to `types::boundary_id` again when we use it, or use a `switch` statement.

Reply:
Ah, drat. Then how about
```
  namespace BoundaryIds
  {
     enum : types::boundary_id {
        a,b,c
     };
  }
```
My goal was just to force using a prefix to ensure that we don't have bare names, but something that indicates the "kind" of object we're referring to.

Reply:
I don't know. Given the fact that we discuss this `enum` now for about 6 months, I have just removed it.

I simply introduce three `constexpr type::boundary_id` in the `Step 69` namespace and use those.